### PR TITLE
Feature #7 - Authentication API

### LINKED ISSUE #7: Authentication
Need to implement functions and corresponding test cases

- [x] Login
- [x] Logout from current session
- [x] Logout from other sessions
- [x] Renew Tokens
- [x] Change Password

### PR DESCRIPTION
**Pull request by hyecheol123**
All features required for Authentication has been implemented

### CODE REVIEW

File: src/datatypes/authentication/ChangePassword.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * @author Hyecheol (Jerry) Jang <hyecheol123@gmail.com>
+ */
+
+/**
+ * Interface for ChangePasswordForm
+ */
+export default interface ChangePasswordForm {
+  currentPassword: string;
+  newPassword: string;
```
Comment (bsstayo) [inline src/datatypes/authentication/ChangePassword.ts:13]: new Password double-check??
As our password policy is quite hard, i think we really need it

File: src/functions/inputValidator/validateChangePasswordForm.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * @author Hyecheol (Jerry) Jang <hyecheo123@gmail.com>
+ */
+
+import Ajv from 'ajv';
+
+export const validateChangePasswordForm = new Ajv().compile({
+  type: 'object',
+  properties: {
+    currentPassword: {type: 'string'},
+    newPassword: {type: 'string'},
```
Comment (bsstayo) [inline src/functions/inputValidator/validateChangePasswordForm.ts:13]: add password double-check here?
Since it's a part of validation...

**Review by bsstayo — COMMENTED** [review]
Reviewed WITHOUT TEST CODE
Leave a simple comment - password double check

File: src/functions/inputValidator/validateChangePasswordForm.ts (hunk 2 above)
Comment (hyecheol123) [reply src/functions/inputValidator/validateChangePasswordForm.ts:13]: This is to check whether response body "contains" newPassaord and currentPassword.

File: src/datatypes/authentication/ChangePassword.ts (hunk 1 above)
Comment (hyecheol123) [reply src/datatypes/authentication/ChangePassword.ts:13]: This is the "Interface", indicating the datatype of password change form. As typescript only checks data type during compile time, we need to implement a separated runtime type checker.

**Comment by hyecheol123** [comment]
@bsstayo it does not double checked... :(

**Review by bsstayo — APPROVED** [review]

**Pull request merged**